src/index.ts

```diff
@@ -150,35 +166,97 @@ function publishStatementInMultiMode (splitOutput: SqlStatement[], currentStatem
         lastSplitResult.value += LINE_FEED
       }
       lastSplitResult.value += currentStatement.value + SEMICOLON
+      if (includePositions) {
+        lastSplitResult.end = currentPos
+      }
     } else {
-      splitOutput.push({
-        value: currentStatement.value + SEMICOLON,
-        supportMulti: true
-      })
+      let stmt: SqlStatement = { value: currentStatement.value + SEMICOLON, supportMulti: true }
+      if (includePositions) {
+        stmt.start = currentStatement.start
+        stmt.end = currentPos
+      }
+      splitOutput.push(stmt)
     }
   } else {
-    splitOutput.push({
-      value: currentStatement.value,
-      supportMulti: false
-    })
+    let stmt: SqlStatement = { value: currentStatement.value, supportMulti: false }
+    if (includePositions) {
+      stmt.start = currentStatement.start
+      stmt.end = currentPos
+    }
+    splitOutput.push(stmt)
   }
 }
 
 function publishStatement (context: SplitExecutionContext): void {
   const trimmed = context.currentStatement.value.trim()
   if (trimmed !== '') {
     if (!context.multipleStatements) {
-      context.output.push({
-        value: trimmed,
-        supportMulti: context.currentStatement.supportMulti
-      })
+      if (context.includePositions) {
+        context.output.push({
+          value: trimmed,
+          supportMulti: context.currentStatement.supportMulti,
+          start: context.currentStatement.start,
+          end: context.position
+        })
+      } else {
+        context.output.push({
+          value: trimmed,
+          supportMulti: context.currentStatement.supportMulti
+        })
+      }
     } else {
       context.currentStatement.value = trimmed
-      publishStatementInMultiMode(context.output, context.currentStatement)
+      if (context.includePositions) {
+        context.currentStatement.end = context.position
+      }
+      publishStatementInMultiMode(context.output, context.currentStatement, context.position, context.includePositions)
     }
   }
+  // Reset current statement for the next statement.
   context.currentStatement.value = ''
   context.currentStatement.supportMulti = true
+  if (context.includePositions) {
+    context.currentStatement.start = context.position
+  }
+}
+
+export type SqlStatementResult = string | { stmt: string, start: number, end: number };
+
+export function split (sql: string, options?: SplitOptions): SqlStatementResult[] {

```

Comment:
Please avoid unnecessary reordering of functions in feature or bugfix PRs. These changes can complicate code reviews, as the diffs become harder to follow.

It would be better to keep the code in its original position and create a separate PR for any rearrangements.